### PR TITLE
avahi: cleanup

### PR DESCRIPTION
**Pull request by jtojnar**
###### Description of changes

- avahi: format the expression
- avahi: drop intltool & outdated flags
- avahi: Simplify pkg-config cross fix
- avahi: simplify path handling

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 173913` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>17 packages marked as broken and skipped:</summary>
  <ul>
    <li>balsa</li>
    <li>birdfont</li>
    <li>cambalache</li>
    <li>font-manager</li>
    <li>gnvim</li>
    <li>gnvim-unwrapped</li>
    <li>libinsane</li>
    <li>lispPackages.cl-webkit2</li>
    <li>lispPackages.nyxt</li>
    <li>luakit</li>
    <li>newsflash</li>
    <li>pdfpc</li>
    <li>rednotebook</li>
    <li>setzer</li>
    <li>webkitgtk</li>
    <li>webkitgtk_4_1</li>
    <li>wike</li>
  </ul>
</details>
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>ario</li>
    <li>fx_cast_bridge</li>
    <li>geoclue2-with-demo-agent</li>
    <li>ola</li>
    <li>rpiplay</li>
    <li>uxplay</li>
  </ul>
</details>
<details>
  <summary>14 packages built:</summary>
  <ul>
    <li>avahi</li>
    <li>avahi-compat</li>
    <li>geoclue2</li>
    <li>mpd</li>
    <li>perl532Packages.ImageSane</li>
    <li>perl534Packages.ImageSane</li>
    <li>pidgin</li>
    <li>purple-matrix</li>
    <li>python310Packages.avahi</li>
    <li>python310Packages.sane</li>
    <li>python39Packages.avahi</li>
    <li>python39Packages.sane</li>
    <li>sane-backends</li>
    <li>shairplay</li>
  </ul>
</details>

File: pkgs/development/libraries/avahi/default.nix [hunk 1]
```diff
@@ -26,45 +34,75 @@ stdenv.mkDerivation rec {
...
+    dbus-python
+  ]);
+
+  configureFlags = [
+    "--disable-gdbm"
+    "--disable-mono"
+    # Use non-deprecated path https://github.com/lathiat/avahi/pull/376
+    "--with-dbus-sys=${placeholder "out"}/share/dbus-1/system.d"
+    (lib.enableFeature gtk3Support "gtk3")
+    (lib.enableFeature qt4Support "qt4")
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/avahi/default.nix:84]:
```suggestion
```

File: pkgs/development/libraries/avahi/default.nix [hunk 2]
```diff
@@ -26,45 +34,75 @@ stdenv.mkDerivation rec {
...
+    expat
+    libiconv
+    libevent
+  ] ++ (with perlPackages; [
+    perl
+    XMLParser
+  ]) ++ lib.optionals gtk3Support [
+    gtk3
+  ] ++ lib.optionals qt4Support [
+    qt4
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/avahi/default.nix:67]:
```suggestion
```

File: pkgs/development/libraries/avahi/default.nix [hunk 3]
```diff
@@ -1,22 +1,30 @@
...
 , qt4Support ? false
 , qt5 ? null
 , qt5Support ? false
 , withLibdnssdCompat ? false
 , python ? null
-, withPython ? false }:
+, withPython ? false
+}:
 
 assert qt4Support -> qt4 != null;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/avahi/default.nix:27]:
```suggestion
```

File: pkgs/development/libraries/avahi/default.nix [hunk 4]
```diff
@@ -1,22 +1,30 @@
...
+, gettext
+, glib
+, libiconv
+, libevent
 , nixosTests
-, gtk3Support ? false, gtk3 ? null
+, gtk3Support ? false
+, gtk3 ? null
 , qt4 ? null
 , qt4Support ? false
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/avahi/default.nix:18]:
```suggestion
```

**Review by SuperSandro2000 — COMMENTED** [review]
Good cleanup!

Qt4 is in the process of being removed if it wasn't already removed. I think this is a good opportunity to also remove it.

File: pkgs/development/libraries/avahi/default.nix (hunk 3 above)
Comment (jtojnar) [inline pkgs/development/libraries/avahi/default.nix:27]:
```suggestion
```

File: pkgs/development/libraries/avahi/default.nix [hunk 5]
```diff
@@ -1,22 +1,30 @@
...
 , qt5 ? null
 , qt5Support ? false
 , withLibdnssdCompat ? false
 , python ? null
-, withPython ? false }:
+, withPython ? false
+}:
 
 assert qt4Support -> qt4 != null;
```
Comment (jtojnar) [inline pkgs/development/libraries/avahi/default.nix:26]:
```suggestion
```

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**